packages/run-protocol/test/psm/test-psm.js

```diff
@@ -0,0 +1,190 @@
+  } = t.context;
+  const { publicFacet } = await E(zoe).startInstance(
+    psmInstall,
+    harden({ AUSD: anchorIssuer }),
+    terms,
+    harden({ feeMintAccess }),
+  );
+  const give = AmountMath.make(anchorBrand, 200n * 1_000_000n);
+  const seat1 = await E(zoe).offer(
+    E(publicFacet).makeSwapInvitation(),
+    harden({ give: { In: give } }),
+    harden({ In: anchorMint.mintPayment(give) }),
+  );
+  const runPayout = await E(seat1).getPayout('Out');
+  const expectedRun = minusFee(give, runBrand);
+  const actualRun = await E(runIssuer).getAmountOf(runPayout);
+  t.deepEqual(actualRun, expectedRun);
+  const liq = await E(publicFacet).getCurrentLiquidity();
+  t.true(AmountMath.isEqual(liq, give));
+  trace('get stable', { give, expectedRun, actualRun, liq });
+  const runGive = AmountMath.make(runBrand, 100n * 1_000_000n);
+  const [runPayment, _moreRun] = await E(runIssuer).split(runPayout, runGive);
+  const seat2 = await E(zoe).offer(
+    E(publicFacet).makeSwapInvitation(),
+    harden({ give: { In: runGive } }),
+    harden({ In: runPayment }),
+  );
+  const anchorPayout = await E(seat2).getPayout('Out');
+  const actualAnchor = await E(anchorIssuer).getAmountOf(anchorPayout);
+  const expectedAnchor = AmountMath.make(
+    anchorBrand,
+    minusFee(runGive, runBrand).value,
+  );
+  t.deepEqual(actualAnchor, expectedAnchor);
+  const liq2 = await E(publicFacet).getCurrentLiquidity();
+  t.deepEqual(AmountMath.subtract(give, expectedAnchor), liq2);
+  trace('get anchor', { runGive, expectedRun, actualAnchor, liq2 });
+});
+
+test('limit', async t => {
+  const {
+    zoe,
+    feeMintAccess,
+    terms,
+    mintLimit,
+    installs: { psmInstall },
+    anchorKit: { brand: anchorBrand, issuer: anchorIssuer, mint: anchorMint },
+  } = t.context;
+  const { publicFacet } = await E(zoe).startInstance(
+    psmInstall,
+    harden({ AUSD: anchorIssuer }),
+    terms,
+    harden({ feeMintAccess }),
+  );
+  const initialPool = AmountMath.make(anchorBrand, 1n);
+  await E(zoe).offer(
+    E(publicFacet).makeSwapInvitation(),
+    harden({ give: { In: initialPool } }),
+    harden({ In: anchorMint.mintPayment(initialPool) }),
+  );
+  t.assert(
+    AmountMath.isEqual(await E(publicFacet).getCurrentLiquidity(), initialPool),
+  );
+  trace('test going over limit');
+  const give = mintLimit;
+  const seat1 = await E(zoe).offer(
+    E(publicFacet).makeSwapInvitation(),
+    harden({ give: { In: give } }),
+    harden({ In: anchorMint.mintPayment(give) }),
+  );
+  trace('gone over limit');
+
+  const paymentPs = await E(seat1).getPayouts();
+  // const refundAmount = await E(localIssuerP).getAmountOf(paymentPs.Transfer);
+
+  trace('PAYOUTS', paymentPs);
+  // We should get 0 RUN but all our anchor back
+  // TODO should this be expecteed to be an empty Out?
+  t.falsy(t.Out);
+  // const actualRun = await E(runIssuer).getAmountOf(runPayout);
+  // t.deepEqual(actualRun, AmountMath.makeEmpty(runBrand));
```

Comment:
please uncomment.

Reply:
This is related to the TODO.  I would love to uncomment, but then it fails because `getAmountOf` fails internally. That's because `t.Out` is undefined (as in line 178). I'm leaving the comment until you and Mark see that and agree that it's expected. Because it surprised me!

Reply:
@erights ^^